core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2125,16 +2128,25 @@ private long startOffsetDuringInitialization(long partitionDataStartOffset) thro
         if (partitionDataStartOffset != PartitionFactory.UNINITIALIZED_START_OFFSET) {
             return partitionDataStartOffset;
         }
-        GroupConfig.ShareGroupAutoOffsetReset offsetResetStrategy;
+        ShareGroupAutoOffsetResetStrategy offsetResetStrategy;
         if (groupConfigManager.groupConfig(groupId).isPresent()) {
             offsetResetStrategy = groupConfigManager.groupConfig(groupId).get().shareAutoOffsetReset();
         } else {
             offsetResetStrategy = GroupConfig.defaultShareAutoOffsetReset();
         }
 
-        if (offsetResetStrategy == GroupConfig.ShareGroupAutoOffsetReset.EARLIEST)
+        if (offsetResetStrategy.timestamp().isEmpty()) {
+            throw new Exception("The timestamp is not available for the share partition: " + topicIdPartition);
```

Comment:
This will translate to UnknownServerError to the client. Do we want to return that or specific exception?

Reply:
I think I would change this to return NoOffsetForPartitionException:
https://github.com/apache/kafka/pull/17972/files#diff-8a7549a9127f9c2eac443cfb35f0d7c1ee8e2884576badd3670a7f19460c59d2R293

Reply:
So, currently, this could only occur when BY_DURATION is used and `duration` somehow is not set. Would IllegalStateException be better to throw as there shouldn't be a case when `offsetResetStrategy.timestamp()` should ideally be empty?